src/serializer/ResidualHeapVisitor.js

```diff
@@ -194,7 +196,7 @@ export class ResidualHeapVisitor {
       visitedBindings[n] = visitedBinding;
     }
     invariant(visitedBinding.value !== undefined);
-    this.visitValue(visitedBinding.value);
+    visitedBinding.value = this.visitEquivalentValue(visitedBinding.value);
```

Comment:
Calls to VisitValue where the value can only sensibly be an object are not candidates for CSE. It turns out that most VisitValue calls are in this category.